### PR TITLE
[1.1] Update buildtools and cli version to same ones as CoreFX

### PR DESCRIPTION
**Pull request by janvorli**
Having separate CLI version greatly increases the cost of adding new
distros, so we should keep coreclr and corefx using the same CLI and
buildtools version for their tools.

Also, this CLI currently being used in coreclr 1.1 is *old*.

Besides copying the files from corefx, I also had to fix an issue
causing the dactablegen.exe, InjectResource.exe and
GenClrDebugResource.exe not being restored. And also an issue that
caused the build to crash when running GenFacades.exe.

### CODE REVIEW

**Comment by janvorli** [comment]
This replaces https://github.com/dotnet/coreclr/pull/17003

File: src/mscorlib/facade/mscorlib.csproj [hunk 1]
```diff
@@ -56,12 +56,11 @@
   <!-- Output paths -->
   <PropertyGroup>
     <BaseIntermediateOutputPath>$(RootBinDir)\obj</BaseIntermediateOutputPath>
-    <!-- Note the trailing '\\' - they are needed so that genfacade does not fail due to an invalid path
-         In particular, the facadepath is sent in quotes.
-         And the IntermediateOutputPath is required to end in a '\' in the coreclr repo. This means that the
-         facadepath ends up being sent as 'mypath\"' and that trips the Path normalization.
-         Introducing a second '\' tricks the normalization logic to treat this '\\"' as '\"' and not complain.    -->
-    <IntermediateOutputPath>$(BaseIntermediateOutputPath)\$(BuildOS).$(BuildArch).$(Configuration)\facade\\</IntermediateOutputPath>
+    <!-- Note the trailing '/' - it is needed so that genfacade does not fail due to an unmatched quote in the
```
Comment (janvorli) [inline src/mscorlib/facade/mscorlib.csproj]: I know this is ugly, but I haven't found any other way to get rid of the trailing backslash that kills the GenFacades.exe tool.

**Comment by janvorli** [comment]
Ah, I've forgotten to update the init-tools.sh too.

**Comment by janvorli** [comment]
@dotnet/dnceng the Ubuntu 14.04 machines that we use for release/1.1.0 CI builds are missing lldb-3.6-dev package which causes the CI to fail for that OS. Could someone please install the missing package?

**Comment by jonfortescue** [comment]
@janvorli Looking into this now -- sorry for getting back so late. This should be an easy fix.

**Comment by jonfortescue** [comment]
@janvorli I have a PR in that should close this issue. Someone on FR will be taking a look at it soon. :)

**Comment by maririos** [comment]
@janvorli we recently installed lldb3.9 in Ubuntu (the changes haven't been rollout) because of a need in CoreFx. Is this somehow going to collide?

FYI @danmosemsft

File: src/mscorlib/facade/mscorlib.csproj (hunk 1 above)
Comment (weshaggard) [reply src/mscorlib/facade/mscorlib.csproj]: This is fine.

**Comment by weshaggard** [comment]
Which version of the init-tool scripts are these changes copied from?

**Comment by janvorli** [comment]
The 3.6 and 3.9 can be installed side by side just fine. The reason for installing 3.6 is that our release/1.1 build depends on it.

**Comment by janvorli** [comment]
@weshaggard 
> Which version of the init-tool scripts are these changes copied from?

This is a copy of the files from the corefx release/1.1.0 repo. So it unifies corefx and coreclr tools in release/1.1.0. There is a small change in the init-tools.sh that adds installing an extra package (Microsoft.DotNet.BuildTools.Coreclr) that coreclr needs and corefx doesn't.

**Comment by joperezr** [comment]
Question: I'm unaware of what is coming on that buildtools update, but will it include using the CLI's MSBuild in order to build non-Windows?

**Comment by janvorli** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test please

**Pull request merged**